Sources/CoreAardvark/PrivateCategories/NSFileHandle+ARKAdditions.m

```diff
@@ -32,8 +46,30 @@
 
 @implementation NSFileHandle (ARKAdditions)
 
++ (void)ARK_setPreventWritesAfterException:(BOOL)preventWritesAfterException
+{
+    // convert from ObjC BOOL (a `char` on some platforms) to stdbool
+    // see: Special Considerations on https://developer.apple.com/documentation/objectivec/bool
+    bool stdboolWritesAfterException = preventWritesAfterException;
+    atomic_store(&__ARKPreventsWritesAfterException, stdboolWritesAfterException);
+
+    if (!stdboolWritesAfterException) {
+        // if we are re-enabling logging, reset state that could prevent writes,
+        // in case we previously encountered an error
+        atomic_store(&__ARKHasEncounteredDiskSizeException, false);
+    }
+}
+
+#pragma mark -
+
 - (void)ARK_writeDataBlock:(NSData *)dataBlock;
 {
+    bool hasEncounteredException = atomic_load(&__ARKHasEncounteredDiskSizeException);
+    bool preventWritesAfterException = atomic_load(&__ARKPreventsWritesAfterException);
+    if (hasEncounteredException && preventWritesAfterException) {
+        return;
+    }
```

Comment:
benchmarks i was looking at (e.g. this AArch64 post https://cpufun.substack.com/p/atomics-in-aarch64) made me not worry about this much

will go with this suggestion since i don't want do dive in further and its reasonable either way.